src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -124,14 +149,19 @@ void close( FileMeta metadata ) throws IOException
                 // if the local lock owner is empty / lock count == 0 for this file, remove it from the node-level locks too.
                 // if the local lock owner's count is NOT 0, then we need to make sure our node's lock-level matches the one in
                 // the local lock manager, and update if necessary
-                LocalLockOwner owner = metadata.getLockOwner();
                 if ( owner == null || owner.getContextLockCount() == 0 )
                 {
-                    // Remove it from node level locks
+                    metadata.removeLock( this.nodeKey );
+                    transactionManager.begin();
+                    metadataCache.put( metadata.getFilePath(), metadata );
+                    transactionManager.commit();
                 }
                 else
                 {
-                    // Compare owner.getLockLevel() with node level lock and update
+                    // Compare owner.getLockLevel() with nodeLevel lock and update
+                    LockLevel nodeLevel = metadata.getLockLevel( this.nodeKey );
+                    // TODO: Negotiate the lock level
+
```

Comment:
Here, I think we know we still have an active lock level, so we will need to check whether the lock level is different from the one in FileMeta, and set / put() if so. If the level actually hasn't changed, we don't want to do the extra cache.put() call.

Also, the ownership change will only happen when the LocalLockOwner.unlock() is called, which I think is after this close() method gets called. We need to process the node-level lock update in the updateDominantLocks() method instead of here.

Reply:
I think this can be resolved, since it's the close() method that I think we're saying should be removed in favor of updateDominantLocks()